lib/data/odata.js

```diff
@@ -129,30 +130,45 @@ const submissionToOData = (fields, table, submission, options = {}) => new Promi
-  const root = Object.assign({ __id: submission.instanceId }, base);
-
+  const root = Object.assign({ }, base);
+
+  // The instanceId and submission metadata we decorate onto every root row based
+  // on option.metadata even though they are not part of the form's own schema.
+  // So rather than try to inject them into the xml transformation below, we just
+  // formulate them herein advance:
+  if (!options.metadata || options.metadata.includes('__id')) {
+    root.__id = submission.instanceId;
+  }
   if (table === 'Submissions') {
-    // the instanceId and submission metadata we decorate onto every root row even
-    // though they are not part of the form's own schema. so rather than try to
-    // inject them into the xml transformation below, we just formulate them here
-    // in advance:
-    const submitterId = (submitter.id == null) ? null : submitter.id.toString(); // => string for future-proofing
-    Object.assign(root, {
-      __system: {
-        submissionDate: submission.createdAt,
-        updatedAt: submission.updatedAt,
-        submitterId,
-        submitterName: submitter.displayName || null,
-        attachmentsPresent: attachment.present || 0,
-        attachmentsExpected: attachment.expected || 0,
-        status: encrypted ? (encHasData ? 'notDecrypted' : 'missingEncryptedFormData') : null,
-        reviewState: submission.reviewState || null,
-        deviceId: submission.deviceId || null,
-        // because of how we compute this value we don't need to default it to 0:
-        edits: submission.aux.edit.count,
-        formVersion: submission.aux.exports.formVersion
+    const systemObj = {
+      submissiondate: { key: 'submissionDate', value: submission.createdAt },
+      updatedat: { key: 'updatedAt', value: submission.updatedAt },
+      submitterid: { key: 'submitterId', value: (submitter.id == null) ? null : submitter.id.toString() }, // => string for future-proofing
+      submittername: { key: 'submitterName', value: submitter.displayName || null },
+      attachmentspresent: { key: 'attachmentsPresent', value: attachment.present || 0 },
+      attachmentsexpected: { key: 'attachmentsExpected', value: attachment.expected || 0 },
+      status: { key: 'status', value: encrypted ? (encHasData ? 'notDecrypted' : 'missingEncryptedFormData') : null },
+      reviewstate: { key: 'reviewState', value: submission.reviewState || null },
+      deviceid: { key: 'deviceId', value: submission.deviceId || null },
+      edits: { key: 'edits', value: submission.aux.edit.count },
+      formversion: { key: 'formVersion', value: submission.aux.exports.formVersion },
+    };
+
+    if (options.metadata) {
+      const systemProp = options.metadata.filter(p => p.startsWith('__system')).map(p => p.substr(9));
+      if (systemProp.length > 0) {
+        root.__system = {};
+        for (const property of systemProp) {
+          root.__system[systemObj[property].key] = systemObj[property].value;
+        }
```

Comment:
I think the user could specify `$select=__system` to get all the data in `__system`: in OData, `__system` is itself a property that is a `ComplexType`.

Reply:
Added this feature

Reply:
will add the behavior for other complexType in a separate PR. It's quite complex

Reply:
added this in https://github.com/getodk/central-backend/pull/717